sensor/common/compliance/node_inventory_handler_test.go

```diff
@@ -47,6 +49,82 @@ func fakeNodeInventory(nodeName string) *storage.NodeInventory {
 	return msg
 }
 
+func fakeNodeIndex(arch string) *v4.IndexReport {
+	return &v4.IndexReport{
+		HashId:  fmt.Sprintf("sha256:%s", strings.Repeat("a", 64)),
+		Success: true,
+		Contents: &v4.Contents{
+			Packages: []*v4.Package{
+				{
+					Id:      "0",
+					Name:    "openssh-clients",
+					Version: "8.7p1-38.el9",
+					Kind:    "binary",
+					Source: &v4.Package{
+						Name:    "openssh",
+						Version: "8.7p1-38.el9",
+						Kind:    "source",
+						Source:  nil,
+						Cpe:     "cpe:2.3:*:*:*:*:*:*:*:*:*:*:*",
+					},
+					PackageDb:      "sqlite:usr/share/rpm",
+					RepositoryHint: "hash:sha256:f52ca767328e6919ec11a1da654e92743587bd3c008f0731f8c4de3af19c1830|key:199e2f91fd431d51",
+					Arch:           arch,
+					Cpe:            "cpe:2.3:*:*:*:*:*:*:*:*:*:*:*",
+				},
+				{
+					Id:      "1",
+					Name:    "openssh-clients-noarch",
+					Version: "8.7p1-38.el9",
+					Kind:    "binary",
+					Source: &v4.Package{
+						Name:    "openssh",
+						Version: "8.7p1-38.el9",
+						Kind:    "source",
+						Source:  nil,
+						Cpe:     "cpe:2.3:*:*:*:*:*:*:*:*:*:*:*",
+					},
+					PackageDb:      "sqlite:usr/share/rpm",
+					RepositoryHint: "hash:sha256:f52ca767328e6919ec11a1da654e92743587bd3c008f0731f8c4de3af19c1830|key:199e2f91fd431d51",
+					Arch:           "noarch",
+					Cpe:            "cpe:2.3:*:*:*:*:*:*:*:*:*:*:*",
+				},
+				{
+					Id:      "2",
+					Name:    "openssh-clients-empty-arch",
+					Version: "8.7p1-38.el9",
+					Kind:    "binary",
+					Source: &v4.Package{
+						Name:    "openssh",
+						Version: "8.7p1-38.el9",
+						Kind:    "source",
+						Source:  nil,
+						Cpe:     "cpe:2.3:*:*:*:*:*:*:*:*:*:*:*",
+					},
+					PackageDb:      "sqlite:usr/share/rpm",
+					RepositoryHint: "hash:sha256:f52ca767328e6919ec11a1da654e92743587bd3c008f0731f8c4de3af19c1830|key:199e2f91fd431d51",
+					Arch:           "",
+					Cpe:            "cpe:2.3:*:*:*:*:*:*:*:*:*:*:*",
+				},
```

Comment:
[nit] It took me a minute to look at all fields and figured out which ones were changed. Maybe it would be good to add a function to create these package structures to make it easier to distinguish hardcoded fields vs params. WDYT?